includes/Query.php

```diff
@@ -2227,18 +2232,25 @@ private function _usedby( $option ) {
 	 * @param mixed $option
 	 */
 	private function _uses( $option ) {
-		$this->addTable( 'templatelinks', 'tl' );
+		$this->addTables( [
+			'linktarget' => 'lt',
+			'templatelinks' => 'tl',
+		] );
+
 		$where = $this->tableNames['page'] . '.page_id=tl.tl_from AND (';
```

Comment:
Need to join templatelinks and linktarget (as introduced in L2235-L2238) or the result will become cross join of these two
```suggestion
		$where = $this->tableNames['page'] . '.page_id=tl.tl_from AND lt.lt_id = tl.tl_target_id AND (';
```